src/scripts/dot-gov.js

```diff
@@ -0,0 +1,363 @@
+    false
+  );
+
+  helpMessage.registerInteractive(
+    ".Gov",
+    "Curious if a (city, state, tribal, judicial...) domain exists (for X)? Charlie can help you find out!",
+    "(entity) .gov (search term)",
+    true
+  );
+
+  /** Exposed only for testing. */
+  module.exports.gitGovRegex = gitGovRegex;
+
+  /**
+   * Transform a search term into a slack message.
+   *
+   * This function obtains a copy of the list of currently registered
+   * .gov domains in CSV format, then filters them by an (optional)
+   * `entity`, then filters them by an (optional) search term, and
+   * finally, formats the results as a series of slack messages.
+   *
+   * Those messages are passed to `say` for display to the user.
+   */
+  module.exports.dotGov = async ({ entity, searchTerm, say, thread }) => {
+    // fetch the domain list
+    const domains = await cache("dotgov domains", 1440, async () =>
+      axios
+        .get(CISA_DOTGOV_DATA_URL)
+        .then((response) => {
+          if (response && response.data) {
+            return parse(response.data, { columns: true });
+          }
+          return [];
+        })
+        .catch(() => [])
+    );
+
+    let filteredDomains = [];
+    let headerText = null;
+    let output = [];
+    const blocks = [];
+
+    const searchTermExists = typeof searchTerm === "string" && searchTerm;
+    const entityExists = typeof entity === "string" && entity;
+
+    // if requested, confine the search to a particular governmental body
+    if (entityExists) {
+      filteredDomains = narrowResultsByEntity(entity, domains);
+    } else {
+      filteredDomains = domains;
+    }
+
+    // if a search term is given, see if we can find a match
+    if (searchTermExists) {
+      const filterByField = (field) =>
+        filterBy(field, searchTerm, filteredDomains);
+      filteredDomains = Array.from(
+        new Set([
+          ...filterByField(DATA_FIELDS.NAME),
+          ...filterByField(DATA_FIELDS.ORG),
+          ...filterByField(DATA_FIELDS.AGENCY),
+          ...filterByField(DATA_FIELDS.CITY),
+          ...filterByField(DATA_FIELDS.STATE),
+          ...filterByField(DATA_FIELDS.TYPE),
+        ])
+      );
+    }
+
+    const numResults = filteredDomains.length;
+
+    // pluralize
+    const domainText = entityExists ? `${entity} domains` : "domains";
+
+    // format a salutation
+    if (numResults > 0 && searchTermExists) {
+      headerText = `Here's what I could find for ${domainText} related to "${searchTerm}".`;
+      const numToDisplay = numResults > 5 ? 5 : numResults;
+      output = selectDomainsAtRandom(filteredDomains, numToDisplay);
+    } else if (numResults > 0) {
+      headerText = `There are ${numResults} \`.gov\` ${domainText} right now! Have you seen this one?`;
+      output = selectDomainsAtRandom(filteredDomains, 1);
+    } else if (numResults === 0 && searchTermExists) {
+      headerText = `I found nothing related to "${searchTerm}", sorry.`;
+    }
```

Comment:
_thinks for a bit_

No, I think this is right, because if there is no search term and also no results, it means there was a server error. As written, Charlie doesn't respond at all in that case.